File: completed/2966.java
import java.util.*;

class Solution {
    public int[][] divideArray(int[] nums, int k) {
        int n = nums.length;
        if (n % 3 != 0) {
            return new int[][] {};
        }

        n /= 3;
        Arrays.sort(nums);
        int[][] matrix = new int[n][3];
        for (int i = 0; i < n; i++) {
            matrix[i][0] = nums[i * 3];
            for (int j = 1; j < 3; j++) {
                if (nums[i * 3 + j] - matrix[i][0] > k) {
                    return new int[][] {};
                }
                matrix[i][j] = nums[i * 3 + j];
            }
        }
        return matrix;
    }
}

File: completed/2966.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        int n = nums.size();
        vector<vector<int>> divided(n / 3, vector<int>(3, 0));

        sort(nums.begin(), nums.end());
        for (int i = 0; i < n; i += 3) {
            if (nums[i + 2] - nums[i] > k) {
                return vector<vector<int>>();
            }

            divided[i / 3][0] = nums[i];
            divided[i / 3][1] = nums[i + 1];
            divided[i / 3][2] = nums[i + 2];
        }
        return divided;
    }
};
